### ISSUE TITLE
TransFire & TransGrays colormaps don't work on scalars

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#611]
## 🐛 Bug

The `TransFire` and `TransGrays` colormaps slightly break the vispy colormap API when `vispy.BaseColormap.__getitem__()` is called with a scalar for example:

```python
In [1]: from napari.util.colormaps import colormaps

In [2]: inferno = colormaps.vispy_or_mpl_colormap('inferno')

# colormaps should be able to take a scalar and return a ColorArray
In [3]: inferno[.5]
Out[3]: <ColorArray: 1 color ((0.732796, 0.21433249, 0.332053, 1.0))>

# but transfire throws an error
In [4]: tfire = colormaps.TransFire()                                                                                                                                               
In [5]: tfire[0.5]
```
```                                                                                                                                                       
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-17-af8bd54a5e2c> in <module>
----> 1 tfire[0.5]

~/miniconda3/envs/napdev/lib/python3.7/site-packages/vispy/color/colormap.py in __getitem__(self, item)
    291         item = np.clip(item, 0., 1.)
    292         colors = self.map(item)
--> 293         return ColorArray(colors)
    294 
    295     def __setitem__(self, item, value):

~/miniconda3/envs/napdev/lib/python3.7/site-packages/vispy/color/color_array.py in __init__(self, color, alpha, clip, color_space)
    146 
    147         # Parse input type, and set attribute"""
--> 148         rgba = _user_to_rgba(color, clip=clip)
    149 
    150         if alpha is not None:

~/miniconda3/envs/napdev/lib/python3.7/site-packages/vispy/color/color_array.py in _user_to_rgba(color, expand, clip)
     55     color = np.atleast_2d(color).astype(np.float32)
     56     if color.shape[1] not in (3, 4):
---> 57         raise ValueError('color must have three or four elements')
     58     if expand and color.shape[1] == 3:  # only expand if requested
     59         color = np.concatenate((color, np.ones((color.shape[0], 1))),

ValueError: color must have three or four elements

```

I think the issue is related to the [`np.dstack`  in `TransFire.map()`](https://github.com/napari/napari/blob/85e6bd46a2b1be230cceef429316e41405aa6356/napari/util/colormaps/colormaps.py#L257) that returns an array with an extra dimension when called on a scalar, causing the ValueError shown above:

```python
In [6]: tfire.map(np.array([1]))                                                                                                                                                    
Out[6]: array([[[1., 1., 1., 1.]]], dtype=float32)
              ^
In [7]: inferno.map(np.array([1]))                                                                                                                                                  
Out[7]: array([[0.98836201, 0.99836397, 0.64492399, 1.        ]])

```

In most cases, this is not a problem, since the colormaps are almost always applied to 3D numpy arrays and never indexed directly.  But one strategy for applying gamma (#610) would require that standard `cm[scalar]` API to work.  I'll try to see if I can tweak the `TransFire.map()` function to work for all cases.

**Comment by sofroniewn** [comment]
Thanks for catching that!

**Comment by tlambert03** [comment]
fixed in #622
